assets/src/date/test/getDateObjectWithTimezone.js

```diff
@@ -0,0 +1,342 @@
+    });
+
+    expect(date).toBe('2020-07-12T04:30:00-07:00');
+  });
+
+  it('should return 2020-07-12T04:30:00-07:00 when gmtOffset is -7 (America/Los_Angeles)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -7,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T04:30:00-07:00');
+  });
+
+  it('should return 2020-07-12T07:30:00-04:00 when timezone is set to America/New_York', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -4,
+      timezone: 'America/New_York',
+    });
+
+    expect(date).toBe('2020-07-12T07:30:00-04:00');
+  });
+
+  it('should return 2020-07-12T07:30:00-04:00 when gmtOffset is -4 (America/New_York)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -4,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T07:30:00-04:00');
+  });
+
+  it('should return 2020-07-12T04:30:00-07:00 when timezone is set to America/Tijuana', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -7,
+      timezone: 'America/Tijuana',
+    });
+
+    expect(date).toBe('2020-07-12T04:30:00-07:00');
+  });
+
+  it('should return 2020-07-12T04:30:00-07:00 when gmtOffset is -7 (America/Tijuana)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -7,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T04:30:00-07:00');
+  });
+
+  it('should return 2020-07-12T08:30:00-03:00 when timezone is set to America/Sao_Paulo', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -3,
+      timezone: 'America/Sao_Paulo',
+    });
+
+    expect(date).toBe('2020-07-12T08:30:00-03:00');
+  });
+
+  it('should return 2020-07-12T08:30:00-03:00 when gmtOffset is -3 (America/Sao_Paulo)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: -3,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T08:30:00-03:00');
+  });
+
+  it('should return 2020-07-12T15:30:00+04:00 when timezone is set to Asia/Dubai', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: 4,
+      timezone: 'Asia/Dubai',
+    });
+
+    expect(date).toBe('2020-07-12T15:30:00+04:00');
+  });
+
+  it('should return 2020-07-12T15:30:00+04:00 when gmtOffset is + 4 (Asia/Dubai)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: 4,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T15:30:00+04:00');
+  });
+
+  it('should return 2020-07-12T19:30:00+08:00 when timezone is set to Asia/Hong_Kong', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: 8,
+      timezone: 'Asia/Hong_Kong',
+    });
+
+    expect(date).toBe('2020-07-12T19:30:00+08:00');
+  });
+
+  it('should return 2020-07-12T19:30:00+08:00 when gmtOffset is + 8 (Asia/Hong_Kong)', () => {
+    const date = getDateObjectWithTimezone({
+      gmtOffset: 8,
+      timezone: '',
+    });
+
+    expect(date).toBe('2020-07-12T19:30:00+08:00');
+  });
```

Comment:
The way these tests repeat it seems that you could set up an array of `timezones` and their associated `gmtOffset` and loop through it to reproduce all these tests.  Something like this:
```
const timezones = {
  'Asia/Hong_Kong': 8,
  'Asia/Dubai': 4,
  ...
}

Object.keys(timezones).forEach((timezone) => {
  it(
    `should return 2020-07-12T19:30:00${getOffset(timezones[timezone])}` when timezone is set to ${timezone}`,
    () => {
      const date = getDateObjectWithTimezone({
        gmtOffset: timezones[timezone],
        timezone,
      });

      expect(date).toBe(`2020-07-12T19:30:00${getOffset(timezones[timezone])}`);
    }
  );

  it(
    `should return 2020-07-12T19:30:00${getOffset(timezones[timezone])}` when gmtOffset is ${timezones[timezone]} (${timezone})`,
    () => {
      const date = getDateObjectWithTimezone({
        gmtOffset: timezones[timezone],
        timezone: '',
      });

      expect(date).toBe(`2020-07-12T19:30:00${getOffset(timezones[timezone])}`);
    }
  );
  
});
```

Not sure if this would make things hard to read, but if it seems good it'll mean we could easily add in a new timezone to test and viola, it'll get the appropriate test.

**NOTE:** The `getOffset` function is just a made up placeholder that would take the gmtOffset and return the string equivalent it needs to be (e.g. 8 => +8:00, -3 => -3:00).

Reply:
i love this idea, adding all of these were very annoying!